### ISSUE TITLE
Always redownload Codeium when reopening VSCode, is this normal?

### ISSUE THREAD

**Issue by feri-irawan** [Exafunction/codeium#39]
Is this normal? Every time I reopen VSCode, Codeium is always redownloading

![shot-2024-02-07_21-03-47](https://github.com/Exafunction/codeium/assets/57158078/1c8d3697-d87d-4c5f-9333-1eacb6409beb)

**Comment by pqn** [comment]
Does Codeium always work after the download? We update versions pretty often, is it possible that your extension has been upgrading on reloads?

**Comment by Beetroit** [comment]
I also experience this. Every single time. It seems the downloaded file is stored in a temp directory. It keeps redownloading.

**Comment by feri-irawan** [comment]
Oh, I see. I think it's a bug because VSCode always downloads Codeium in the same size, 25 MB. Codeium works well after downloading.

**Comment by luckyalade** [comment]
I am experiencing the same issue. It keeps on redownloading whenever I restart my vscode.

**Comment by pqn** [comment]
For anyone experiencing this, can you include the logs in the Codeium tab in the output pane in VS Code? The logs may help indicate if the download file isn't being found at all, or if the expected checksum of the file is incorrect after download, which causes a redownload. Also reporting this in a new issue since this one has been closed by the original author.

**Comment by teknosains** [comment]
I'am also having the same issue. I use VSCode on Mac

**Comment by zengjunke** [comment]
I want to know if this problem has been solved? I have also encountered such problems

**Comment by feri-irawan** [comment]
> Does Codeium always work after the download? We update versions pretty often, is it possible that your extension has been upgrading on reloads?

@zengjunke

**Comment by zengjunke** [comment]
How often does the software update? I'm not sure if I download it again every time I open it, but I feel like almost every time I open vscode, I download it again.
&nbsp;


***@***.***
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Feri ***@***.***&gt;; 
发送时间: 2024年5月31日(星期五) 下午2:10
收件人: ***@***.***&gt;; 
抄送: ***@***.******@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Exafunction/codeium] Always redownload Codeium when reopening VSCode, is this normal? (Issue #39)














  
Does Codeium always work after the download? We update versions pretty often, is it possible that your extension has been upgrading on reloads?
  
@zengjunke

**Comment by nufair2006** [comment]
I think this has been solved as it only installs whenever there is a new update version. However, is there a way to prevent updating to the newer version frequently? I don't want to install every new version and I want to be able to update to a newer version only once a month as this saves me data and other inconsistencies. if this functionality comes up, the issue will be most likely gone.

**Comment by supervova** [comment]
@pqn 

> For anyone experiencing this, can you include the logs in the Codeium tab in the output pane in VS Code?

(VSCode) 2024-07-21 08:29:38.372 [INFO]: mac-ca initialized
(VSCode) 2024-07-21 08:29:43.282 [INFO]: No proxy settings found
(VSCode) 2024-07-21 08:29:58.938 [ERROR]: Failed to initialize Unleash: undefined
(VSCode) 2024-07-21 08:30:08.695 [INFO]: Creating manager directory: /var/folders/wm/xbchyhb16d9_dtsl_n512_z40000gn/T/5e4c4764-a944-44b3-9a3f-fef34b332f88/codeium/manager
(VSCode) 2024-07-21 08:30:08.701 [INFO]: 1e20fa38ede961073c4494bc9637bffe13e6a8f4 not found, downloading language server
(VSCode) 2024-07-21 08:30:09.868 [INFO]: No proxy settings found
(VSCode) 2024-07-21 08:35:12.685 [ERROR]: Failed to download from https://github.com/Exafunction/codeium/releases/download/language-server-v1.8.83/language_server_macos_arm.gz
(VSCode) 2024-07-21 08:35:12.694 [ERROR]: Language server binary not found: undefined

**Comment by abel1502** [comment]
Why is this closed? It isn't fixed yet, I am experiencing the same problem

**Comment by Vengxing** [comment]
Anyone found solution to this problem yet? because I'm sure it downloads everytime not because it updates everytime, because there's no way it updates every few minutes reopen. And why is this thread closed?

**Comment by feri-irawan** [comment]
Maybe we need to **disable auto update**.

**Comment by Domingos-Daniel** [comment]
I had the same problem, but I've downloaded manually a server version in https://github.com/Exafunction/codeium/releases, and extraced the .exe file and placed in C:\Users\PC\.vscode\extensions\codeium.codeium-1.12.6\dist, i am using **windows**, then just restart your vs code and it's done, hope it was useful .

**Comment by HanslettTheDev** [comment]
It's surprising that this issue was closed meanwhile it's still a problem

**Comment by streetlife** [comment]
I stopped closing VSCode. Restarting VSCode always forced a download of the language server, and if I have a connection issue, I will need to restart VSCode.

**Comment by CRIMSON-CORP** [comment]
This is still an issue, please reopen, i still have this issue

**Comment by starlingvibes** [comment]
I still have this issue and it's becoming pretty annoying tbh

**Comment by WerdienJihed** [comment]
I have been experiencing this issue for months now. And It's 100% not at update It can be triggered on window reload more than once a day. 
The download size is 31.55MB each time.

**Comment by HanslettTheDev** [comment]
After some investigation, I found some two unique errors that should give a hint to what is actually going wrong
```cmd
Codeium: ENOENT: no such file or directory, rename *c:\Users\DELL\,vscode\extensions\codeium.codeium- 1.23.4\dist\d7e8f56c3ee3982e7019ba6161979a54e8f0b89
O\language_server_windows_x64.exe.download' -> *c:\Users\DELL\vscode\extensions\codeium.codeium- 1.23.4\dist\d7e8f56c3ee3982e7019ba6161979a54e8f0b89
O\language_server_windows_x64.exe' If issues persist after restarting your IDE, please contact vscode@codeium.com. Attempted download URL:
https://releases.codeiumdata.com/language-server- v1.22.4/language_server_windows_x64.exe.gz.
X
Source: Codeium: Al Coding Autocomplete and Chat for Python, Javascript. Type...
```

And this other error
![image](https://github.com/user-attachments/assets/dcc7f7e1-7cf7-4a20-8822-5995101ae996)

Based on my findings and looking at the vscode exthost.log file and the extension.js file, the issue could be related to CRC32.
**What is CRC32?**
"CRC32 stands for Cyclic Redundancy Check 32-bit and is used to detect accidental changes to raw data in digital networks and storage devices. It's used for error-checking to ensure data integrity"

And after taking a look at the extension.js(Almost 2.5 million lines of minified JS Code), I realized they could be an issue how this check is done that somehow triggers the redownload issue. Typically when the CRC32 check fails, Codeium attempts to redownload the language server exe file into this location **C:\Users\[Username]\.vscode\extensions\codeium.codeium-1.23.4\dist**. It's possible for a file to get downloaded but the CRC32 check fails and therefore causes this continuous cycle of redownloading the file

**Temporal fix**
You could manually download the language server file or make a copy of it and store it in the same location so codeium always find it

But I guess the maintainers can look into this better. Since I don't think the vscode extension has been made public yet!!

**Comment by HanslettTheDev** [comment]
You can use the vscode developer tools to see more output from codeium when vscode tries activating extensions and gets to codeium

**Comment by Mehdi-Messaadi-BUYSELL** [comment]
I can confirm that this issue persists on my end to this day. Would greatly appreciate any fix for this.

**Comment by yubaoquan** [comment]
I have several vscode windows open, and each vscode window will download codeium when I reload it. I think it should only download once.

**Comment by teknosains** [comment]
it seems still haven't been fixed yet.

**Comment by teknosains** [comment]
Dec 2024 still persists

**Comment by Esensats** [comment]
Issue still persists on Android Studio 2024.2.2. Plus it takes around 20 minutes to download with 0.5 mbps speed, even though my internet speed is 200+ mbps normally. Caching a single file should be such a non-issue, usually. Well I guess the code was written by an LLM ¯\_(ツ)_/¯

**Comment by umardev500** [comment]
> Issue still persists on Android Studio 2024.2.2. Plus it takes around 20 minutes to download with 0.5 mbps speed, even though my internet speed is 200+ mbps normally. Caching a single file should be such a non-issue, usually. Well I guess the code was written by an LLM ¯_(ツ)_/¯

same to me on android lady bug... but the download speed is normal

**Comment by lumenatech** [comment]
Same issue still persists on android studio
